### PR TITLE
[sdk/python] Generate pyproject.toml

### PR DESCRIPTION
**Pull request by lblackstone**
### Proposed changes
Generate the pyproject.toml file to enable wheel support for publishing.

### Related issues (optional)
Fix https://github.com/pulumi/pulumi-kubernetes/issues/2431

### CODE REVIEW

**Comment by lblackstone** [comment]
I'm not quite sure what I'm doing here. Does this look right @justinvp @kpitzen?

File: Makefile [hunk 1]
```diff
@@ -82,16 +82,19 @@ nodejs_sdk::
...
 		echo "module fake_python_module // Exclude this directory from Go tools\n\ngo 1.17" > go.mod && \
+		pip3 install wheel && \
 		python3 setup.py clean --all 2>/dev/null && \
 		rm -rf ./bin/ ../python.bin/ && cp -R . ../python.bin && mv ../python.bin ./bin && \
 		sed -i.bak -e 's/^VERSION = .*/VERSION = "$(PYPI_VERSION)"/g' -e 's/^PLUGIN_VERSION = .*/PLUGIN_VERSION = "$(VERSION)"/g' ./bin/setup.py && \
 		rm ./bin/setup.py.bak && \
-		cd ./bin && python3 setup.py build sdist
+		sed -i.bak -e 's/^  version = .*/  version = "$(PYPI_VERSION)"/g' ./bin/pyproject.toml && \
+		rm ./bin/pyproject.toml.bak && \
+		cd ./bin && python3 setup.py sdist bdist_wheel
```
Comment (t0yv0) [inline Makefile:97]: I've played with this; wheels are being built which seems good. The TOML way of building wheels is:

```
python3 --use-pep517 -m pip wheel .
```

It looks like even when using that setup.py is still picked up and runs the auto-install command.

I'm wondering if eventually setup.py needs to be removed.

**Comment by t0yv0** [comment]
I think we can merge this and observe if master branch goes through a dev build  publish to PyPI.

**Comment by t0yv0** [comment]
ci-scripts are currently pushing the source distribution no the wheel:

https://github.com/pulumi/scripts/blob/master/ci/publish-tfgen-package#L114C1-L119C22

without changing that I think adding this toml file here is a no-op

I can follow up with PRs to try publishing the wheel instead.

**Comment by t0yv0** [comment]
Continuing this in https://github.com/pulumi/pulumi-kubernetes/pull/2531

**Pull request closed without merging**